ui/src/app/views/concept-set-details/component.tsx

```diff
@@ -0,0 +1,321 @@
+import {SlidingFabReact} from 'app/views/sliding-fab/component';
+import {Concept, ConceptSet, WorkspaceAccessLevel} from 'generated/fetch';
+
+const styles = reactStyles({
+  conceptSetHeader: {
+    display: 'flex', flexDirection: 'row', justifyContent: 'space-between', paddingBottom: '1.5rem'
+  },
+  conceptSetTitle: {
+    color: '#2F2E7E', fontSize: 20, fontWeight: 600, marginBottom: '0.5rem',
+    display: 'flex', flexDirection: 'row'
+  },
+  conceptSetMetadataWrapper: {
+    flexDirection: 'column', alignItems: 'space-between', marginLeft: '0.5rem'
+  },
+  conceptSetData: {
+    display: 'flex', flexDirection: 'row', color: '#000', fontWeight: 600
+  },
+  buttonBoxes: {
+    color: '#2691D0', borderColor: '#2691D0', marginBottom: '0.3rem'
+  }
+});
+
+const ConceptSetMenu: React.FunctionComponent<{
+  disabled: boolean, canDelete: boolean, canEdit: boolean, onEdit: Function, onDelete: Function
+}> = ({disabled, canDelete, canEdit, onEdit, onDelete}) => {
+
+  return <PopupTrigger
+    side='right'
+    closeOnClick
+    content={ <React.Fragment>
+      <TooltipTrigger content={<div>Requires Write Permission</div>}
+                      disabled={canEdit}>
+        <MenuItem icon='pencil'
+                  onClick={() => onEdit}
+                  disabled={!canEdit}>
+          Edit
+        </MenuItem>
+      </TooltipTrigger>
+      <TooltipTrigger content={<div>Requires Owner Permission</div>}
+                      disabled={canDelete}>
+        <MenuItem icon='trash' onClick={onDelete} disabled={!canDelete}>
+          Delete
+        </MenuItem>
+      </TooltipTrigger>
+    </React.Fragment>}
+  >
+    <Clickable disabled={disabled} data-test-id='workspace-menu'>
+      <ClrIcon shape='ellipsis-vertical' size={21}
+               style={{color: disabled ? '#9B9B9B' : '#216FB4', marginLeft: -9,
+                 cursor: disabled ? 'auto' : 'pointer'}}/>
+    </Clickable>
+  </PopupTrigger>;
+};
+
+export const ConceptSetDetails =
+  fp.flow(withUrlParams(), withCurrentWorkspace())(class extends React.Component<{
+    urlParams: any, workspace: WorkspaceData}, {
+      conceptSet: ConceptSet, deleting: boolean, deletingConcepts: boolean,
+      deleteSubmitting: boolean, editName: string, editDescription: string, editSaving: boolean,
+      error: boolean, errorMessage: string, editing: boolean, loading: boolean,
+      selectedConcepts: Concept[]}> {
+    constructor(props) {
+      super(props);
+      this.state = {
+        conceptSet: undefined,
+        editName: '',
+        editDescription: '',
+        editSaving: false,
+        error: false,
+        errorMessage: '',
+        deleting: false,
+        deletingConcepts: false,
+        deleteSubmitting: false,
+        editing: false,
+        loading: true,
+        selectedConcepts: []
+      };
+    }
+
+    componentDidMount() {
+      this.getConceptSet();
+    }
+
+    async getConceptSet() {
+      const {urlParams: {ns, wsid, csid}} = this.props;
```

Comment:
I think I made this comment to Scout on some non-frontend PR once, but is there a strong reason to use such short variable name abbreviations here? It seems like in this case they're coming directly from the urlParams props, so destructuring into a local variable is super convenient.

But on the other hand, "ns" isn't exactly clear to someone looking at this code for the first time. I'd almost always prefer a real word, even if slightly shortened.

I suspect this is a pretty strong pattern throughout the codebase, so it's not worth breaking consistency to change it here – I just wanted to note that this feels slightly out of place with respect to how we name variables elsewhere.